crypto/s2n_hmac.c

```diff
@@ -179,52 +216,26 @@ int s2n_hmac_init(struct s2n_hmac_state *state, s2n_hmac_algorithm alg, const vo
         S2N_ERROR(S2N_ERR_HMAC_INVALID_ALGORITHM);
     }
 
-    s2n_hash_algorithm hash_alg;
-    state->currently_in_hash_block = 0;
 
-    GUARD(s2n_hmac_hash_alg(alg, &hash_alg));
-    GUARD(s2n_hmac_digest_size(alg, &state->digest_size));
-    GUARD(s2n_hmac_xor_pad_size(alg, &state->xor_pad_size));
+    
+    state->alg = alg;
     GUARD(s2n_hmac_hash_block_size(alg, &state->hash_block_size));
+    state->currently_in_hash_block = 0;
+    GUARD(s2n_hmac_xor_pad_size(alg, &state->xor_pad_size));
+    GUARD(s2n_hmac_digest_size(alg, &state->digest_size));
+
 
     gte_check(sizeof(state->xor_pad), state->xor_pad_size);
     gte_check(sizeof(state->digest_pad), state->digest_size);
-
-    state->alg = alg;
+    gte_check(sizeof(state->xor_pad), state->xor_pad_size);
```

Comment:
This is a duplicate check and can be removed (also at L#228).

Reply:
looks like a git merge FAIL.  I'll fix that.